platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/event/GossipEvent.java

```diff
@@ -332,7 +386,8 @@ public boolean equals(final Object o) {
         }
 
         final GossipEvent that = (GossipEvent) o;
-        return Objects.equals(getHashedData(), that.getHashedData());
+        return Objects.equals(getHashedData(), that.getHashedData())
+                && Objects.equals(consensusData, that.consensusData);
```

Comment:
Not something for this pr.

In production, the only way we should ever compare events is by their hashes. `equals()` and `hashCode()` methods are essentially unit test code that live in production classes. I advocate that we remove them, or perhaps even have them throw an unsupported operation exception. 

We currently only use them in unit tests. But for unit tests, we can implement static comparison methods.

Reply:
> throw an unsupported operation exception.
That would not allow to store that data types in any hash collection.

Reply:
yeah, i agree

Reply:
in fact, we do use these for production. which is why I think I should change this method. we put them in hash sets for production. this is something I need to fix